src/autolabel/labeler.py

```diff
@@ -234,54 +243,63 @@ def run(
                     "Warning: label_selection only supported for classification tasks!"
                 )
             else:
-                self.label_selector = LabelSelector.from_examples(
-                    labels=self.config.labels_list(),
+                self.label_selector = LabelSelector(
+                    config=self.config,
                     embedding_func=PROVIDER_TO_MODEL.get(
                         self.config.embedding_provider(), DEFAULT_EMBEDDING_PROVIDER
-                    )(),
-                    k=self.config.label_selection_count(),
+                    )(model=self.config.embedding_model_name()),
                 )
 
         current_index = self.task_run.current_index if self.create_task else 0
         cost = 0.0
         postfix_dict = {}
 
         indices = range(current_index, len(dataset.inputs))
-
+        selected_labels = self.config.labels_list()
         for current_index in track_with_stats(
             indices,
             postfix_dict,
             total=len(dataset.inputs) - current_index,
             console=self.console,
         ):
             chunk = dataset.inputs[current_index]
+            examples = []
 
-            if self.example_selector:
-                examples = self.example_selector.select_examples(
-                    safe_serialize_to_string(chunk)
-                )
-            else:
-                examples = []
-            # Construct Prompt to pass to LLM
             if (
                 self.config.label_selection()
                 and self.config.task_type() == TaskType.CLASSIFICATION
             ):
-                selected_labels = self.label_selector.select_labels(chunk["example"])
-                final_prompt = self.task.construct_prompt(
-                    chunk,
-                    examples,
-                    selected_labels=selected_labels,
-                    max_input_tokens=self.llm.DEFAULT_CONTEXT_LENGTH,
-                    get_num_tokens=self.llm.get_num_tokens,
-                )
+                # get every column except the one we want to label
+                toEmbed = chunk.copy()
+                del toEmbed[self.config.label_column()]
```

Comment:
what if the label column doesnt exist in the incoming chunk, i.e we are dealing with an unlabeled dataset?